### ISSUE TITLE
json2dcm: incorrect Transfer Syntax in output file

### ISSUE THREAD

**Issue by jimOnAir** [dcm4che/dcm4che#615]
Hello. I have problems with incorrect transfer syntax in some images created with json2dcm.
All these images have US Modality and `1.2.840.10008.1.2.4.50` TransferSyntaxUid. When I am trying to change some dicom tags with the command:

```console
json2dcm  -i source.dcm -j tags.json -o output.dcm
```

the output file has `1.2.840.10008.1.2.4.91` TransferSyntaxUid.
The sample of tags.json:

```JSON
{
  "00080050": { "vr": "SH", "Value": ["68353"] },
  "00080051": {
    "vr": "SQ",
    "Value": [
      {
        "00400031": { "vr": "UT", "Value": ["EXAMPLE"] },
        "00400032": { "vr": "UT", "Value": ["example.com"] },
        "00400033": { "vr": "CS", "Value": ["DNS"] }
      }
    ]
  },
  "00080080": { "vr": "LO", "Value": ["EXAMPLE"] },
  "00380300": { "vr": "LO", "Value": ["AL"] }
}
```

json2dcm versions I've tried: 5.15.1, 5.16.0, 5.19.1, 5.20.0. 

When I explicitly specify TransferSyntaxUid in the command line, everything works well.

```console
json2dcm  -i source.dcm -j tags.json -t 1.2.840.10008.1.2.4.50 -o output.dcm
```

I can't provide a sample of images due to HIPAA, nor reproduce this problem with other images.
Could you please help me to solve this problem?

**Comment by fouadz** [comment]
I wasn't able to reproduce the issue with 5.21.0 using your json tags.

**Comment by gunterze** [comment]
Verify, if the Transfer Syntax of your source image is actual 1.2.840.10008.1.2.4.50. E.g.:
```
$ dcmdump ~/testdata/rubomedical/0002.DCM | grep 0002,0010
248: (0002,0010) UI #22 [1.2.840.10008.1.2.4.50] TransferSyntaxUID
```

**Comment by jimOnAir** [comment]
Thanks for your reply @fouadz, @gunterze. I've checked files and they have several 0002,0010 tags:
```console
./1.2.840.113619.2.435.385050688.1579512017.112.1:
250: (0002,0010) UI #22 [1.2.840.10008.1.2.4.50] TransferSyntaxUID
148342: >>>>(0002,0010) UI #22 [1.2.840.10008.1.2.4.91] TransferSyntaxUID
```
The second one in 6003,1010 sequence
And there is an error in the description. The actual reproduse case is:
1. Create json dump of file:
```console
dcm2json 19B12582 > 19B12582.json
```
2. Modify some tags in json
3. Create patched dicom
```console
json2dcm -i 19B12582 -j 19B12582.json -o 19B12582.patched
```
The diff beetween them excluding bulkData offset is following
```
20c20
<       "1.2.840.10008.1.2.4.50"
---
>       "1.2.840.10008.1.2.4.91"
9442,9447d9441
<                           "00020010": {
<                             "Value": [
<                               "1.2.840.10008.1.2.4.91"
<                             ],
<                             "vr": "UI"
<                           },
9482,9487d9475
<                           "00020010": {
<                             "Value": [
<                               "1.2.840.10008.1.2.4.91"
<                             ],
<                             "vr": "UI"
<                           },
```

**Comment by gunterze** [comment]
(0002,eeee) attributes are not allowed in Data Sets. s. [DICOM PS 3.5](http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_7.1):

> Standard Data Elements have an even Group Number that is not (0000,eeee), (0002,eeee), (0004,eeee), or (0006,eeee).
> 
> Note
> 
> Usage of these groups is reserved for DIMSE Commands (see PS3.7) and DICOM File Formats.
> 

Did your modified json still include the sequence with the items containing them.

**Comment by jimOnAir** [comment]
Yes. Those tags leave unchanged. I need to modify some other tags, for example, those like in the first message.

> (0002,eeee) attributes are not allowed in Data Sets. s. DICOM PS 3.5:

Unfortunately, they are there and I'm nothing to do with it.

**Comment by gunterze** [comment]
Will make JsonReader more robust against incorrect (0002,eeee) in nested data set items - but you also would only need to include the modified attributes in the json.

**Comment by gunterze** [comment]
Fixed by #634
